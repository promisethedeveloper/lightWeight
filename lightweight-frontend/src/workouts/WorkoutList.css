h3 {
	font-weight: 600;
	color: #461111;
}
