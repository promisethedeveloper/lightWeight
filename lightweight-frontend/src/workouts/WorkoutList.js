import React, { useEffect, useState, useContext } from "react";
import { Link } from "react-router-dom";
import UserContext from "../auth/UserContext";
import LightWeightAPI from "../api/api";
import WorkoutDay from "./WorkoutDay";
import { v4 as uuidv4 } from "uuid";

import "./WorkoutList.css";

/** Show all workouts */

const WorkoutList = () => {
	const [workouts, setWorkouts] = useState([]);

	const { currentUser } = useContext(UserContext);
	const { username } = currentUser;

	useEffect(() => {
		getWorkouts(username);
	}, [username]);

	async function getWorkouts(username) {
		try {
			let res = await LightWeightAPI.getWorkouts(username);
			setWorkouts(res);
		} catch (error) {
			console.log(error);
		}
	}

	const onWorkoutDelete = (dayOfWeek) => {
		setWorkouts(workouts.filter((w) => w.dayOfWeek !== dayOfWeek));
	};

	return (
		<div>
			<div className="col-md-8 offset-md-2">
				<h3 className="mb-3 pb-2 border-bottom">WORKOUTS</h3>
				{workouts.length ? (
					<div>
						{workouts.map(({ id, dayOfWeek, description }) => (
							<WorkoutDay
								dayOfWeek={dayOfWeek}
								description={description}
								key={uuidv4()}
								id={id}
								onWorkoutDelete={onWorkoutDelete}
							/>
						))}
					</div>
				) : null}
			</div>
			{workouts.length < 7 && (
				<div className="d-flex flex-row justify-content-center">
					<Link className="p-2 mt-3" to="/workouts/day">
						Add workout day
					</Link>
				</div>
			)}
		</div>
	);
};

export default WorkoutList;
