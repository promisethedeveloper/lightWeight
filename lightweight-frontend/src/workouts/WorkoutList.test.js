import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import WorkoutList from "./WorkoutList";
import UserContext from "../auth/UserContext";

/** Smoke test */
it("should render without crashing", () => {
	const currentUser = { username: "John" };
	render(
		<MemoryRouter>
			<UserContext.Provider value={{ currentUser }}>
				<WorkoutList />
			</UserContext.Provider>
		</MemoryRouter>
	);
});

/** Snapshot test */
it("should match snapshots", () => {
	const currentUser = { username: "John" };

	const { asFragment } = render(
		<MemoryRouter>
			<UserContext.Provider value={{ currentUser }}>
				<WorkoutList />
			</UserContext.Provider>
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});

it("should render form label text", () => {
	let currentUser = { username: "Promise" };
	const { getByText } = render(
		<MemoryRouter>
			<UserContext.Provider value={{ currentUser }}>
				<WorkoutList />
			</UserContext.Provider>
		</MemoryRouter>
	);
	expect(getByText("WORKOUTS")).toBeInTheDocument();
	expect(getByText("Add workout day")).toBeInTheDocument();
});
