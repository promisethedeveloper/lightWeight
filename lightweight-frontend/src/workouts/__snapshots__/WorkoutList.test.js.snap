// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshots 1`] = `
<DocumentFragment>
  <div>
    <div
      class="col-md-8 offset-md-2"
    >
      <h3
        class="mb-3 pb-2 border-bottom"
      >
        WORKOUTS
      </h3>
    </div>
    <div
      class="d-flex flex-row justify-content-center"
    >
      <a
        class="p-2 mt-3"
        href="/workouts/day"
      >
        Add workout day
      </a>
    </div>
  </div>
</DocumentFragment>
`;
